src/scripts/main.js

```diff
@@ -1,3 +1,263 @@
+  }
+  containerTitle.classList.add('title');
+  containerTitle.innerText = title;
+  containerDescr.innerText = description;
+
+  container.appendChild(containerTitle);
+  container.appendChild(containerDescr);
+  pageBody.appendChild(container);
+
+  setTimeout(() => {
+    container.style.display = 'none';
+  }, 3000);
+};
+
+formButton.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  if (
+    nameInput.value === '' ||
+    positionInput.value === '' ||
+    ageInput.value === '' ||
+    ageInput.value < MIN_AGE ||
+    ageInput.value > MAX_AGE ||
+    salaryInput.value === ''
+  ) {
+    return pushNotification(
+      'Error',
+      'Please, fix all the problems in form',
+      'error',
+    );
+  }
+
+  const newEmployee = document.createElement('tr');
+
+  for (let cell = 0; cell < inputs.length; cell++) {
+    const newCell = document.createElement('td');
+
+    if (cell === 4) {
+      const inputNumber = parseInt(inputs[cell].value);
+
+      newCell.textContent = `$${inputNumber.toLocaleString('en-US')}`;
+    } else {
+      newCell.textContent = inputs[cell].value;
+    }
+
+    newEmployee.appendChild(newCell);
+  }
+
+  inputs.forEach((input, i) => {
+    if (i !== 2) {
+      input.value = '';
+    }
+  });
+
+  tableBody.appendChild(newEmployee);
+
+  return pushNotification('Done', 'New employee was added to list', 'success');
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const editedCell = e.target.closest('td');
+  const cellRow = e.target.closest('tr');
+  const indexOfCell = [...cellRow.children].indexOf(editedCell);
+  const cellInitialtName = editedCell.innerText;
+
+  if (editedCell.firstChild.tagName !== 'FORM') {
+    const edCellForm = document.createElement('form');
+    const edCellLabel = document.createElement('label');
+    const edCellInput = document.createElement('input');
+    const cellInfo = {
+      0: 'Name',
+      1: 'Position',
+      2: 'Office',
+      3: 'Age',
+      4: 'Salary',
+    };
+
+    edCellInput.type = indexOfCell < 3 ? 'text' : 'number';
+    edCellInput.value = cellInitialtName;
+    edCellLabel.appendChild(edCellInput);
+    edCellForm.setAttribute('previous', cellInitialtName);
+    edCellForm.setAttribute('cellType', cellInfo[indexOfCell]);
+    edCellForm.appendChild(edCellLabel);
+
+    editedCell.textContent = '';
+    editedCell.appendChild(edCellForm);
+  }
+});
+
+const onSubmitForm = (e) => {
+  const formHandle = e.target.closest('form');
+
+  if (formHandle) {
+    e.preventDefault();
+
+    const parentElem = formHandle.parentElement;
+    const formAttributes = [...formHandle.attributes];
+    const formPreviousName = formAttributes.find(
+      (el) => el.name === 'previous',
+    );
+    const formCellType = formAttributes.find((el) => el.name === 'celltype');
+    const formCellTypeSalary = formCellType.value === 'Salary';
+    const inputElem = formHandle.querySelector('input');
+
+    formHandle.addEventListener('submit', (ev) => {
```

Comment:
Ensure the form submission is correctly handled without attaching a new event listener each time. We can attach one listener on the form outside of the function.